test/jdk/java/time/test/java/time/TestInstantSource.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.java.time;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertSame;
+import static org.testng.Assert.assertTrue;
+
+import java.lang.reflect.Field;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.InstantSource;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Test instant source.
+ */
+@Test
+public class TestInstantSource {
+
+    private static final ZoneId PARIS = ZoneId.of("Europe/Paris");
+
+    public void test_system() {
+        // main tests for Clock.currentInstant() are in TestClock_System
+        var test = InstantSource.system();
+        assertSame(test.withZone(ZoneOffset.UTC), Clock.systemUTC());
+        assertEquals(test.withZone(PARIS), Clock.system(PARIS));
+        var millis = System.currentTimeMillis();
+        var testMillis = test.millis();
+        var testInstantMillis = test.instant().toEpochMilli();
+        assertTrue(Math.abs(testMillis - millis) < 1000);
+        assertTrue(Math.abs(testInstantMillis - millis) < 1000);
+        assertSame(test.equals(InstantSource.system());
+        assertEquals(test.hashCode(), InstantSource.system().hashCode());
+        assertEquals(test.toString(), "SystemInstantSource");
+    }
+
+    public void test_tick() {
+        var millis = 257265861691L;
+        var instant = Instant.ofEpochMilli(millis);
+        var duration = Duration.ofSeconds(1);
+        var test = InstantSource.tick(InstantSource.fixed(instant), duration);
+        assertEquals(test.withZone(ZoneOffset.UTC), Clock.tick(Clock.fixed(instant, ZoneOffset.UTC), duration));
+        assertEquals(test.withZone(PARIS), Clock.tick(Clock.fixed(instant, PARIS), duration));
+        assertEquals(test.millis(), (millis / 1000) * 1000);
+        assertEquals(test.instant(), instant.truncatedTo(SECONDS));
+        assertSame(test.equals(InstantSource.tick(InstantSource.fixed(instant), duration));
+        assertEquals(test.hashCode(), InstantSource.tick(InstantSource.fixed(instant), duration).hashCode());
+    }
+
+    public void test_fixed() {
+        var millis = 257265861691L;
+        var instant = Instant.ofEpochMilli(millis);
+        var test = InstantSource.fixed(instant);
+        assertEquals(test.withZone(ZoneOffset.UTC), Clock.fixed(instant, ZoneOffset.UTC));
+        assertEquals(test.withZone(PARIS), Clock.fixed(instant, PARIS));
+        assertEquals(test.millis(), millis);
+        assertEquals(test.instant(), instant);
+        assertSame(test.equals(InstantSource..fixed(instant));
+        assertEquals(test.hashCode(), InstantSource..fixed(instant).hashCode());
```

Comment:
Not sure this would compile.

Reply:
I was expecting the GitHub Action to pickup coding and test issues (since my dev setup can't compile or run tests). But it seems it doesn't. do that. The latest commit should at least compile as I copied the test class to another IDE location, but I have no way of knowing if the tests pass.

Reply:
GitHub Actions just verifies build and tier1 testing, while java.time tests are in tier2, so it won't catch the failure. The test case still does not compile without the `import java.time.InstantSource` statement. I've verified the test passed with that fix.